src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -27,7 +30,48 @@ public class SwaggerGenerator : ISwaggerProvider
             _schemaGenerator = schemaGenerator;
         }
 
+        public SwaggerGenerator(
+            SwaggerGeneratorOptions options,
+            IApiDescriptionGroupCollectionProvider apiDescriptionsProvider,
+            ISchemaGenerator schemaGenerator,
+            IAuthenticationSchemeProvider authentiationSchemeProvider) : this(options, apiDescriptionsProvider, schemaGenerator)
+        {
+            _authenticationSchemeProvider = authentiationSchemeProvider;
+        }
+
+        public async Task<OpenApiDocument> GetSwaggerAsync(string documentName, string host = null, string basePath = null)
+        {
+            var (applicableApiDescriptions, swaggerDoc, schemaRepository) = GetSwaggerDocumentWithoutFilters(documentName, host, basePath);
```

Comment:
What's the reasoning for pulling the document filter processing out of the shared method? The first 7 lines of both `GetSwaggerAsync` and `GetSwagger` look identical so just wondering why all of that can't be pulled up into the shared method? And, if there's a good reason, might be good to put a comment there because it's not clear.

Reply:
I did this as a bit of a "premature optimization" in the event that support for async filters is fully pursued per https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1661. I've removed it since it is not explicitly required for this change.